lua-tree/share/lua/5.1/kong/tools/utils.lua

```diff
@@ -1237,5 +1257,13 @@ do
   end
 end
 
+-- No reason to yield a Lua coroutine based on `ngx.get_phase()`,
+-- as we're not running this within Nginx anyway.
+_M.yield = function() return end
+
+local time_ns = function()
+    return require("socket").gettime() * 10^9
```

Comment:
Where is the source for this gettime function?
I couldn't grep my way to it.

Reply:
Ah, it's actually part of a the Lua socket C-lib: https://github.com/lunarmodules/luasocket/blob/5b18e475f38fcf28429b1cc4b17baee3b9793a62/src/timeout.c#L36

Guessing this function is unused during the code execution (however it was previously failing due to the imports before). I'll switch it to use something else.

Reply:
If nothing for plugin validation is using `utils.time_ns()` then we can safely remove this function and add that to the patch. It looks like all the imports are localized so removing should be sufficient.

Reply:
Indeed, nothing is directly using it right now. As it's a global function though, technically speaking couldn't custom plugin schemas use it for "insert any reason"? It wasn't difficult to support, so I was personally leaning to support it. Sure, it doesn't return the exact time in nanoseconds (and technically that could be a security risk if one were using the value for randomness seeding), but for our current use-case of plugin validation, I felt this was okay.

I'm perfectly fine with removing it if that's what we want, however, I'm just thinking about these things from a maintainability perspective and "what if" it's used in the future, because it's a global function afterall.

Reply:
> technically speaking couldn't custom plugin schemas use it for "insert any reason"?

Being a `util` and not built into the PDK I think we can safely assume that no custom plugins will be using this in their custom validations.

> I'm just thinking about these things from a maintainability perspective and "what if" it's used in the future, because it's a global function after all.

Fair point, but I think for those what if future uses we can deal with things then or provide them a solution that works within their schema.

Reply:
You do make some good points there too. I honestly could go either way here. I'll assume we'll just remove the global `time_ns()` function unless anyone says otherwise. Removing it could break things in a future Kong Lua code update, however, the chances are that are likely rare.